src/metabase/task/truncate_audit_tables.clj

```diff
@@ -0,0 +1,115 @@
+  :visibility :internal
+  :setter     :none
+  :audit      :never
+  :getter     (fn []
+                (let [env-var-value (setting/get-value-of-type :integer :audit-max-retention-days)]
+                  (def env-var-value env-var-value)
+                  (cond
+                    (nil? env-var-value)
+                    default-retention-days
+
+                    ;; Treat 0 as an alias for infinity
+                    (zero? env-var-value)
+                    ##Inf
+
+                    (< env-var-value min-retention-days)
+                    (do
+                      (log-minimum-value-warning env-var-value)
+                      min-retention-days)
+
+                    :else
+                    env-var-value))))
+
+(defn- truncate-table!
+  "Given a model, deletes all rows older than the configured threshold"
+  [model time-column]
+  (when-not (infinite? (audit-max-retention-days))
+    (let [table-name (name (t2/table-name model))]
+      (task-history/with-task-history {:task "task-history-cleanup"}
+        (try
+          (log/infof "Cleaning up %s table" table-name)
+          (let [rows-deleted (t2/delete!
+                              model
+                              time-column
+                              [:<= (t/minus (t/offset-date-time) (t/days (audit-max-retention-days)))])]
+            (if (> rows-deleted 0)
+              (log/infof "%s cleanup successful, %d rows were deleted" table-name rows-deleted)
+              (log/infof "%s cleanup successful, no rows were deleted" table-name)))
+          (catch Throwable e
+            (log/errorf e "%s cleanup failed" table-name)))))))
+
+(defenterprise audit-models-to-truncate
+  "List of models to truncate. OSS implementation only truncates `query_execution` table."
+  metabase-enterprise.task.truncate-audit-tables
+  []
+  [{:model :model/QueryExecution :timestamp-col :started_at}])
+
+(defn- truncate-audit-tables!
+  []
+  (run!
+   (fn [{:keys [model timestamp-col]}]
+     (truncate-table! model timestamp-col))
+   (audit-models-to-truncate)))
+
+(jobs/defjob ^{:doc "Triggers the removal of `query_execution` rows older than the configured threshold."} TruncateAuditTables [_]
+  (truncate-audit-tables!))
+
+(def ^:private truncate-audit-tables-job-key "metabase.task.truncate-audit-tables.job")
+(def ^:private truncate-audit-tables-trigger-key "metabase.task.truncate-audit-tables.trigger")
+(def ^:private truncate-audit-tables-cron "0 0 */12 * * ? *")
+
+(defmethod task/init! ::TruncateAuditTables [_]
+  (let [job     (jobs/build
+                 (jobs/of-type TruncateAuditTables)
+                 (jobs/with-identity (jobs/key truncate-audit-tables-job-key)))
+        trigger (triggers/build
+                 (triggers/with-identity (triggers/key truncate-audit-tables-trigger-key))
+                 (triggers/start-now)
+                 (triggers/with-schedule
+                   (cron/schedule
+                    ;; run every 12 hours
```

Comment:
Nit: would this comment be better above line 101, so we don't forget to update the comment if we update the cron?